addons/website_sale/views/templates.xml

```diff
@@ -1510,25 +1510,31 @@
-                                <h3 class="mb24">Pay with </h3>
-                                <t t-call="payment.payment_tokens_list">
-                                    <t t-set="mode" t-value="'payment'"/>
-                                    <t t-set="submit_txt">Pay Now</t>
-                                    <t t-set="icon_right" t-value="1"/>
-                                    <t t-set="icon_class" t-value="'fa-chevron-right'"/>
-                                    <t t-set="submit_class" t-value="'btn btn-primary'"/>
-                                    <t t-set="pms" t-value="tokens"/>
-                                    <t t-set="form_action" t-value="'/shop/payment/token'"/>
-                                    <t t-set="prepare_tx_url" t-value="'/shop/payment/transaction/'"/>
-                                    <t t-set="partner_id" t-value="partner"/>
-
-                                    <t t-set="back_button_icon_class" t-value="'fa-chevron-left'"/>
-                                    <t t-set="back_button_txt">Return to Cart</t>
-                                    <t t-set="back_button_class" t-value="'btn btn-secondary'"/>
-                                    <t t-set="back_button_link" t-value="'/shop/cart'"/>
-                                </t>
-                            </div>
+                            <t t-if="website_sale_order.amount_total">
+                                <div id="payment_method" class="mt-3" t-if="acquirers or tokens">
+                                    <h3 class="mb24">Pay with </h3>
+                                    <t t-call="payment.payment_tokens_list">
+                                        <t t-set="mode" t-value="'payment'"/>
+                                        <t t-set="submit_txt">Pay Now</t>
+                                        <t t-set="icon_right" t-value="1"/>
+                                        <t t-set="icon_class" t-value="'fa-chevron-right'"/>
+                                        <t t-set="submit_class" t-value="'btn btn-primary'"/>
+                                        <t t-set="pms" t-value="tokens"/>
+                                        <t t-set="form_action" t-value="'/shop/payment/token'"/>
+                                        <t t-set="prepare_tx_url" t-value="'/shop/payment/transaction/'"/>
+                                        <t t-set="partner_id" t-value="partner"/>
+
+                                        <t t-set="back_button_icon_class" t-value="'fa-chevron-left'"/>
+                                        <t t-set="back_button_txt">Return to Cart</t>
+                                        <t t-set="back_button_class" t-value="'btn btn-secondary'"/>
+                                        <t t-set="back_button_link" t-value="'/shop/cart'"/>
+                                    </t>
+                                </div>
+                                <div t-else="" class="alert alert-warning">
```

Comment:
```suggestion
                                <div t-if="not (acquirers or tokens)" class="alert alert-warning">
```
I think in stable there is a good chance that someone did a xpath over `<div id="payment_method">` so it might be more safe to just do another t-if (eg. if someone had an xpath adding something after `id=payment_method`, the new t-else would cause an error).

If we do that, it might also make sense to not add an indentation.